### ISSUE TITLE
could not find function "sg.smooth"

### ISSUE THREAD

**Issue by mpmarcaida** [jeffreyevans/spatialEco#19]
I get this error: Error in `sg.smooth(t(...  : could not find function "sg.smooth"` when I run my code using Windows computer. The "spatialEco" package is installed.. Checked the version and it says 1.1-0. I originally wrote my code in my Mac and I didn't experience any glitch. I noticed the package version in my Mac and it says 1.2-0. Could this be causing the problem? How come they have different versions when I just did the same process of `install.packages()`? 

Thanks!

**Comment by jeffreyevans** [comment]
Without more information I really cannot say what is going on here. Especially since the current CRAN version is at 1.3-0. The function is available at both 1.3-0 and 1.2-0 so, it should be showing up. I first uninstall the version that you currently have and then I would try reinstalling from CRAN and then check the version to make sure it is current.

uninstall.packages("spatialEco", .Library)
install.packages("spatialEco")

You can then check the package version:
packageVersion("spatialEco")

or you can install the development version from GitHub. First install the “devtools” package and then use:
remotes::install_github("jeffreyevans/spatialEco")

If you are still having issues please let me know. Please keep in mind that package dependences and changes in R can create a complex set of circumstances where a previous package version is installed. It is important to keep your version of R, along with packages, up to date. If you are running an old versions of R this could explain why you are getting an old package version with install.packages.

Best,
Jeff

**Comment by mpmarcaida** [comment]
Hi Jeff, 

Thanks for the quick reply. I did just what you said, uninstalled and reinstalled spatialEco but the version is still 1.1-0. I tried installing the Github version and I got this error:

```
ERROR: this R is version 3.5.1, package 'spatialEco' requires R >=  3.6.0
* removing '~/AppData/Local/Continuum/anaconda3/envs/rstudio/lib/R/library/spatialEco'
In R CMD INSTALL
Warning messages:
1: In missing_devel_warning(pkgdir) :
  Package spatialEco has compiled code, but no suitable compiler(s) were found. Installation will likely fail.
  Install Rtools and make sure it is in the PATH.
2: In i.p(...) :
  installation of package ‘~/AppData/Local/Temp/RtmpYLOtLH/remotes192055d248d4/jeffreyevans-spatialEco-8de9d92’ had non-zero exit status
```

**Comment by jeffreyevans** [comment]
Two things here, you are not running a current version of R. It is at 3.6.2 and I am requiring 3.6.0 for installing the latest version of the package. Without updating your R version to >=3.6 you can only install spatialEco 1.2-0. The sg_smooth function is available at that version but not at 1.1-0. With you running R 3.5.1 there is no reason that you should be getting spatialEco 1.1-0 in your install unless one of the package dependencies is way out of date. Try updating your libraries (menu option  packages > update packages) and then reinstalling spatialEco from CRAN. But that aside, I would highly recommend updating your R version and then updating all of your packages.

Also, you do not have the RTools software installed. This is necessary for installing packages that require compiling from source. While RTools is not necessary, it is if you want to compile and install packages from source. Most of what is on CRAN are the precompiled binaries but, it is not uncommon that a new uncompiled version is sitting on CRAN while it is waiting for processing and distribution to all of the mirrors. This process can sometimes take a few weeks and in the meantime, without RTools you have no way to install the most current version of the package. You can get the RTools install here: https://cran.r-project.org/bin/windows/Rtools/index.html

Best,
Jeff

**Comment by mpmarcaida** [comment]
Alright! Thanks so much!
